src/main/java/org/mifos/ops/zeebe/camel/routes/OperationsRouteBuilder.java

```diff
@@ -23,6 +27,61 @@
     @Override
     public void configure() {
 
+        /**
+         * Bulk cancellation of active process by processId
+         *
+         * method: [PUT]
+         * request body: {
+         *     processId: [123, 456, 789]
+         * }
+         *
+         * response body: {
+         *     success: [], # list of processId which was successfully cancelled
+         *     failed: [] # list of processId whose cancellation wasn't successful
+         *     cancellationSuccessful: int # total number of process which was successfully cancelled
+         *     cancellationFailed: int # total number of process whose cancellation wasn't successful
+         *
+         * }
+         */
+        from("rest:PUT:/channel/workflow")
+                .id("bulk-cancellation")
+                .log(LoggingLevel.INFO, "## bulk cancellation by process id")
+                .process(exchange -> {
+
+                    JSONObject object = new JSONObject(exchange.getIn().getBody(String.class));
+                    JSONArray processIds = object.getJSONArray("processId");
+
+                    JSONArray success = new JSONArray();
+                    JSONArray failed = new JSONArray();
+
+                    AtomicInteger successfullyCancelled = new AtomicInteger();
+                    AtomicInteger cancellationFailed = new AtomicInteger();
+
+
+                    processIds.forEach(elm -> {
+                        long processId = Long.parseLong(elm.toString());
+
+                        try {
+                            zeebeClient.newCancelInstanceCommand(processId).send().join();
+                            success.put(processId);
+                            successfullyCancelled.getAndIncrement();
+                        }catch (Exception e) {
+                            failed.put(processId);
+                            cancellationFailed.getAndIncrement();
+                        }
```

Comment:
Might be worthwhile to log the error as well.

Reply:
done